main.c

```diff
@@ -3872,6 +3873,20 @@ ovl_setattr (fuse_req_t req, fuse_ino_t ino, struct stat *attr, int to_set, stru
             }
           node->ino->mode = mode = st.st_mode;
         }
+      ret = faccessat(dirfd, node->path, R_OK|W_OK, AT_SYMLINK_NOFOLLOW);
+      if (ret < 0)
+        {
+          if (errno == EACCES)
+            {
+              need_restore_mode = true;
+              ret = fchmodat(dirfd, node->path, node->ino->mode | S_IRUSR | S_IWUSR, 0);
+            }
+          if (ret < 0)
+            {
+              fuse_reply_err (req, errno);
+              return;
+            }
+        }
```

Comment:
it is an extra cost that is not necessary in most cases, especially when running in a user namespace.

Could we just detect the failure from `safe_open` and work directly on  the path when that happens?  If the file could not be opened, could we attempt directly `fchmodat(dirfd, node->path, desired_mode)` and similarly for other functionalities?